src/main/webapp/threadHandler_test.js

```diff
@@ -0,0 +1,82 @@
+// TODO(alicevlasov) figure out how to run tests without copying function
+function hasValidCharacters(name) {
+  var regex = /^[a-zA-Z0-9-_]+$/;
+  return regex.test(name);
+}
+
+var assert = require('assert');
+
+describe('#hasValidCharacters()', function() {
+  const expectedOutput = true;
+  const lowerCase = "abcdefghijklmnopqrstuvwxyz";
+  const upperCase = lowerCase.toUpperCase();
+  const numbers = "1234567890";
+  const dashUnderscore = "-_";
+
+  describe('should return false when all characters are valid', function() {
```

Comment:
Do you mean should return true

Reply:
Whoops yes that's what I meant to write!